org/all-prs.ts

```diff
@@ -1,31 +1,72 @@
 import { schedule, danger, warn, fail } from "danger"
-const pr = danger.github.pr
 
 const isJest = typeof jest !== "undefined"
 
-const rfc = isJest ? require("./utils").rfc : (id: string, reason: string, closure: any) => closure()
+// Stores the parameter in a closure that can be invoked in tests.
+const storeRFC = (reason: string, closure: () => void | Promise<any>) => {
+  // We return a closure here so that the (promise is resolved|closure is invoked) 
+  // during test time and not when we call rfc().
+  return () => {
+    return closure instanceof Promise ? closure : Promise.resolve(closure())
+  }
+}
+
+// Either schedules the promise for execution via Danger, or invokes closure.
+const runRFC = (reason: string, closure: () => void | Promise<any>) =>
+  closure instanceof Promise ? schedule(closure) : closure()
+
+const rfc: any = isJest ? storeRFC : runRFC
 
 import yarn from "danger-plugin-yarn"
-rfc("1", "Highlight package dependencies on Node projects", () => {
-  schedule(yarn())
+rfc("Highlight package dependencies on Node projects", async () => {
```

Comment:
I like this because it kind of abstracts out how we build+test RFCs. They can be closures, or async closures, and you test both of them the same way.

Reply:
This doesn't work I'm afraid, it'll work in tests, but not in production, an async function has to be called with schedule: http://danger.systems/js/guides/the_dangerfile.html#async

Reply:
If we move fake the `schedule` call, it can still probably be tested fine though 👍

Reply:
Right, so when `rfc` gets called for actual PRs, then [`schedule` gets called](https://github.com/artsy/artsy-danger/pull/21/files#diff-bb4cdb0982ea1e4edde39e85e5ff7bccR16) (other the closure is immediately invoked). So I _think_ it should work?

Reply:
Hah - yeah, that should work - great spot.

Reply:
I'll merge this when I can watch for potential incoming PR breakages